br/pkg/lightning/restore/restore.go

```diff
@@ -2245,15 +2240,21 @@ func (cr *chunkRestore) deliverLoop(
 		// Update the table, and save a checkpoint.
 		// (the write to the importer is effective immediately, thus update these here)
 		// No need to apply a lock since this is the only thread updating `cr.chunk.**`.
-		// In local mode, we should write these checkpoint after engine flushed.
+		// In local mode, we should write these checkpoints after engine flushed.
 		cr.chunk.Checksum.Add(&dataChecksum)
 		cr.chunk.Checksum.Add(&indexChecksum)
 		cr.chunk.Chunk.Offset = currOffset
 		cr.chunk.Chunk.PrevRowIDMax = rowID
+		// currOffset may be less than endOffset even if we have read all rows of this chunk.
```

Comment:
when will this happen? seems a bug

Reply:
```
insert into t(a) values (1, 2, 3);
```
`currOffset` will be 
```
insert into t(a) values (1,2,3);
                              ^
                              |
```

Reply:
In other words, reading all rows isn't equivalent to reading all bytes of a chunk.

Reply:
encode loop may return error and then this channel is closed, in this case we should not do `cr.chunk.Chunk.Offset = cr.chunk.Chunk.EndOffset`, `channelClosed` is not enough to cover this case.

Reply:
Yes, you are right. 

`encodeLoop` will send a nil slice to `kvsCh` after reading all rows. `deliverLoop` can distinguish between channel closed and rows are all read.

Reply:
Fixed. PTAL again, thanks.